### DISCUSSION TITLE
Configure Message Box configuration items via the command line

### DISCUSSION THREAD

**Discussion by Adrian-Rasmussen** [WinMerge/winmerge#2223] [Q&A] [answered]
When attempting to run WinMerge in the background, particularly with `/noprefs /noninteractive /minimize` the comparison can be halted by the "You are about to compare very large files" message box, requiring user input to continue. 

Is there an option to skip this message via the command line? Trying to use `/cfg MessageBoxes/CompareLargeFiles=6"` gives `Invalid key 'MessageBoxes/CompareLargeFiles' specified in /config option`

**Comment by sdottaka** [comment, marked as answer]
Unfortunately, there are some key names that cannot be specified in the command line option `/cfg`, and `MessageBoxes/CompareLargeFiles` is one of them.

The command line option `/noprefs` specifies that settings are not read from the registry. One workaround is to pre-populate the message box answers in the MessageBoxes category of WinMerge's Options window, remove this command line option, and run WinMerge with the current settings.

Alternatively, start WinMerge to refer to the INI file as shown below.
~~~
"c:\Program Files\winmerge\WinMergeU.exe" /inifile c:\tmp\winmerge.ini
~~~
When you configure MessageBoxes and other settings as you like in the Options window, etc., and close WinMerge, the settings will be saved to the specified INI file instead of the registry.

By referencing this INI file as shown below and running WinMerge in the background, you can run it without being affected by the settings saved in the current registry.

~~~
"c:\Program Files\winmerge\WinMergeU.exe" /noninteractive /minimize /inifile c:\tmp\winmerge.ini c:\dev\winmerge\src d:\dev\winmerge\src /or c:\tmp\ report.html /cfg IncludeFileCmpReport=1 /cfg ReportType=2
~~~
